src/shared_modules/content_manager/src/components/executionContext.hpp

```diff
@@ -47,9 +93,23 @@ class ExecutionContext final : public AbstractHandler<std::shared_ptr<UpdaterBas
             std::filesystem::create_directories(databasePath);
         }
 
-        // Create a RocksDB instance
+        // Inititalize RocksDB driver instance.
         context.spRocksDB = std::make_unique<Utils::RocksDBWrapper>(databasePath + databaseName);
-        context.spRocksDB->put(Utils::getCompactTimestamp(std::time(nullptr)), "0");
+
+        // Read input offsets.
+        const auto databaseOffset {getDatabaseOffset(context)};
+        const auto configOffset {getConfigOffset(context.configData)};
+
+        // Choose the greatest between the DB and the config offset.
+        const auto currentOffset {std::max(databaseOffset, configOffset)};
+
+        if (currentOffset != databaseOffset)
+        {
+            // Put the current offset in the database.
+            context.spRocksDB->put(Utils::getCompactTimestamp(std::time(nullptr)), std::to_string(currentOffset));
+        }
```

Comment:
Improve: To better reflect the intent

> If the offset is set on the database, the chosen offset is the greatest between the one in the database and the one in the input config.
 
```suggestion
        if (currentOffset > databaseOffset)
        {
            // Put the current offset in the database.
            context.spRocksDB->put(Utils::getCompactTimestamp(std::time(nullptr)), std::to_string(currentOffset));
        }
```

Reply:
What I want to reflect with this equal condition is that, if `currentOffset` and `databaseOffset` are equal, there is no need to insert the "new" offset in the database, given that it is the same as the last already inserted.

Reply:
Exactly. The `lessThan` scenario should not happen, if that's your concern.

Reply:
Question: Would it be the same as doing if `(currentOffset != databaseOffset || currentOffset == 0)` and removing thr spRocksDB-put from getDatabaseOffset ?

Reply:
> Would it be the same as doing if (currentOffset != databaseOffset || currentOffset == 0) and removing the spRocksDB-put from getDatabaseOffset ?

Yes... and no :sweat_smile: 
If the offset is zero and it doesn't change between executions, it will be inserted multiple times.
This scenario is possible if, for example, you are using the `S3Downloader` that doesn't modify the offset.

An alternative I thought about was to have some kind of `firstExecution` flag (set when `getDatabaseOffset()` throws) that could be used instead of `currentOffset == 0`.

Reply:
@sebasfalcone finally I've changed the condition as you said.